src/NHibernate/Util/CollectionHelper.cs

```diff
@@ -579,6 +579,77 @@ public IEnumerator GetEnumerator()
 			#endregion
 		}
 
+		/// <summary>
+		/// A read-only generic wrapper around ICollection interface
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		[Serializable]
+		private class CollectionWrapper<T> : ICollection<T>
+		{
+			private readonly ICollection _col;
+
+			public CollectionWrapper(ICollection col)
+			{
+				_col = col;
+			}
+
+			public int Count => _col.Count;
+
+			public bool IsReadOnly => true;
+
+			public bool Contains(T item)
+			{
+				foreach (var obj in _col)
```

Comment:
Shouldn't it just call `_col.Contains(item)`?

Reply:
There is no ICollection.Contains. Though could be _col.Cast\<T\>().Contains. Should I made this change?

Reply:
No. Thats ok.